Documents/Documents/Code/Controllers/Documents/ASCDocumentsViewController.swift

```diff
@@ -1303,14 +1303,28 @@ class ASCDocumentsViewController: ASCBaseTableViewController, UIGestureRecognize
             showLoadingPage(false)
             showEmptyView(false)
 
-            if !ASCNetworkReachability.shared.isReachable {
-                errorView?.type = .networkError
-            } else {
-                errorView?.type = .error
-            }
-
-            if let error = error {
-                errorView?.subtitleLabel?.text = "\(errorView?.subtitleLabel?.text ?? "") (\(error.localizedDescription))"
+            if let error = error as? NetworkingError {
+                switch error {
+                case let .apiError(error):
+                    if let onlyofficeError = error as? OnlyofficeServerError {
+                        switch onlyofficeError {
+                        case .paymentRequired:
+                            errorView?.type = .paymentRequired
+                            errorView?.onAction = {
+                                if let url = URL(string: OnlyofficeApiClient.shared.baseURL?.absoluteString ?? ASCConstants.Urls.applicationPage) {
+                                    UIApplication.shared.open(url, options: [:], completionHandler: nil)
+                                }
+                            }
+                        default:
+                            errorView?.type = .error
```

Comment:
Should add a detailed description of the error from the api of onlyoffice?
errorView?.subtitleLabel?.text = "\(errorView?.subtitleLabel?.text ?? "") (\(error.localizedDescription))"

Reply:
Fix commited